packages/cli/src/databases/types.ts

```diff
@@ -0,0 +1,27 @@
+import type { Logger } from '@/Logger';
+import type { QueryRunner } from 'typeorm';
+
+export type DatabaseType = 'mariadb' | 'postgresdb' | 'mysqldb' | 'sqlite';
+
+export interface MigrationContext {
+	logger: Logger;
+	queryRunner: QueryRunner;
+	tablePrefix: string;
+	dbType: DatabaseType;
+	dbName: string;
+	migrationName: string;
+}
+
+type MigrationFn = (ctx: MigrationContext) => Promise<void>;
+
+export interface MigrationInterface {
+	transaction?: boolean;
```

Comment:
I cannot quite tell what is this for? Also, regardless of the purpose, instead of an optional `boolean` we could type it to the non-default `false` instead, to avoid clutter like `transaction = true` in future.